components/eamxx/src/mct_coupling/scream_cxx_f90_interface.cpp

```diff
@@ -57,10 +57,15 @@ void fpe_guard_wrapper (const Lambda& f) {
   try {
     f();
   } catch (std::exception &e) {
```

Comment:
Any reason not to change this to `catch (...)` for greater generality?

Reply:
Yeah, we can do that

Reply:
Ah, but I want to print the exception message.

Reply:
I'll do a rethrow, then add `catch(...)` underneath

Reply:
Does that work? Or do you need to nest try blocks?

Reply:
I think you need to nest try-catch like this:

```c++
  try {
    try {
      throw std::exception();
    } catch (const std::exception& e) {
      printf("hi\n");
      throw;
    }
  } catch (...) {
    printf("hello\n");
  }
```

Reply:
I'm fine with this, but you might consider nesting try-catch blocks to avoid redundant code:
```c++
  try {
    try {
      throw std::exception();
    } catch (const std::exception& e) {
      printf("hi\n");
      throw;
    }
  } catch (...) {
    printf("hello\n");
  }
```